clients/java/src/test/java/io/camunda/zeebe/client/ZeebeClientTest.java

```diff
@@ -728,6 +738,125 @@ public void shouldNotSetDefaultJobWorkerTenantIdsFromSetterWithCloudClientBuilde
         .isEqualTo(builder);
   }
 
+  @Test
+  @DisplayName("should throw when gatewayTarget is invalid")
+  public void shouldThrowWhenGatewayTargetIsInvalid() {
+    // given
+    final ZeebeClientBuilderImpl builder = new ZeebeClientBuilderImpl();
+
+    // when
+    Assertions.assertThrows(RuntimeException.class, () -> builder.gatewayTarget("0.0.0.0:26500"));
+  }
+
+  @Test
+  @DisplayName("should use gatewayAddress when gatewayAddress is valid ip:port")
+  public void shouldUseGatewayAddressWhenGatewayAddressIsValidIpPort() {
+    // given
+    final ZeebeClientBuilderImpl builder = new ZeebeClientBuilderImpl();
+    final Properties properties = new Properties();
+    properties.putIfAbsent(ClientProperties.GATEWAY_ADDRESS, "0.0.0.0:26500");
+    builder.withProperties(properties);
+
+    // when
+    builder.build();
+
+    // then
+    Assertions.assertNull(builder.getGatewayTarget());
+
+    Assertions.assertEquals("0.0.0.0:26500", builder.getGatewayAddress());
+  }
+
+  @Test
+  @DisplayName("should use gatewayTarget when gatewayAddress is not valid ip:port")
+  public void shouldUseGatewayTargetWhenGatewayAddressIsNotValidIpPort() {
+    // given
+    final ZeebeClientBuilderImpl builder = new ZeebeClientBuilderImpl();
+    final Properties properties = new Properties();
+    properties.putIfAbsent(ClientProperties.GATEWAY_ADDRESS, "gateway.zeebe.com:26500");
+
+    // use
+    builder.withProperties(properties);
+
+    // verify
+    Assertions.assertEquals("gateway.zeebe.com:26500", builder.getGatewayTarget());
+  }
```

Comment:
:x: We shouldn't have this behavior. The `GATEWAY_ADDRESS` client property shouldn't set the `gatewayTarget`.

Reply:
> We can actually set the` GATEWAY_ADDRES` client property at a Invalid value now, should I add a validation for this situation to make sure the `GATEWAY_ADDRES` property is a valid value.

HI @koevskinikola 

I have changed my code but got a questions above. Could you help me to see that? Thank you!